### PR TITLE
Adding python 2.7 compatibility fixes

### PR DESCRIPTION
**Pull request by Yurmusp**
Enabling Python 2.7 in build and tests.

### CODE REVIEW

File: minio_storage/storage.py [hunk 1]
```diff
@@ -28,10 +30,10 @@ class MinioStorage(Storage):
     """
     file_class = ReadOnlySpooledTemporaryFile
 
-    def __init__(self, minio_client, bucket_name, *,
```
Comment (thomasf) [inline minio_storage/storage.py]: Oh, this is a python3.6 thing.. py36 only allows named arguments after the *(args) argument so it's a calling saftey thing.. Just remove it for python2 compatiblity.

**Comment by thomasf** [comment]
This is how tox.ini should be modified:

```diff
diff --git a/tox.ini b/tox.ini
index ee56cf6..08f17d7 100644
--- a/tox.ini
+++ b/tox.ini
@@ -7,6 +7,7 @@ python_files = tests.py test_*.py *_tests.py
 
 [travis]
 python =
+       2.7: py27
        3.4: py34
        3.5: py35, lint, isort, docs
        3.6: py36
@@ -58,7 +59,7 @@ commands = mkdocs build
 
 [tox]
 envlist =
-       {py34,py35}-django{18,19,110,111}-minio222,
+       {py27,py34,py35}-django{18,19,110,111}-minio222,
        py36-django111-minio{,222},
        lint,
        isort,
```

When you have that (or even before) you should be able to run the py27 tests locally using tox:

```sh
tox -e py27-django111-minio222
```

File: tests/test_app/tests/utils.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 import hashlib
 import os
+import six
```
Comment (thomasf) [inline tests/test_app/tests/utils.py]: Please avoid adding a six dependency, use `django.utils.six` if you want to use it.

Comment (thomasf) [inline tests/test_app/tests/utils.py]: I have no idea of they intend to remove that package in django 2.0.. If it's going to be removed it's probably better to just copy the stuff from six that is needed into a local six.py module or something like that.

**Comment by Yurmusp** [comment]
OK. Will take into account.
I got a bit stuck with unicode.
Did request to base library maintainers and now wait for this pull request.
https://github.com/minio/minio-py/pull/530#discussion_r120465332
Will continue when guys are done. They promise it will be there soon.

Thinks work to me in my dirty version if used only with ASCII.
To have official change I have to wait for this unicode change
as tests do unicode everywhere.

**Comment by Yurmusp** [comment]
@thomasf ,

Please, review. 
This pull requests enables base python 2.7 build with checks similar to python 3.4.
With latest update of minio 2.2.3 we have unicode fix from minio team
that I requested and tests now pass.
I need this to be merged. So that I could add other changes on top of this. 
Further changes are not related to python 2.7. 
They going to be just general logic fixes that are not related to specific python.
Therefore I want those as different PR.

**Comment by thomasf** [comment]
looks good..

use the vendored version of six from django.utils.six  and sqash all commits into one.

**Comment by Yurmusp** [comment]
Just looked at it.
Django has it starting version 1.10, while we are going to work down to 1.8.
So, either have to be conditional import or just use common six package.
Just use common will be less ugly.

**Comment by Yurmusp** [comment]
Actually, no. they have it in 1.8 as well. OK. Will do.

**Comment by Yurmusp** [comment]
@thomasf ,
I think it is ready. 
It is 2 commits because of conflict. I can not squash conflicting commit from master in my branch.
I think you can do Squash and Merge  instead of just Merge for PR and that shall give you just one commit.

**Pull request merged**